src/theme/colour-primitive/theme-helper.ts

```diff
@@ -0,0 +1,125 @@
+import { get } from "lodash";
+import {
+    ColourCollectionsMap,
+    ColourScheme,
+    ColourSet,
+    ThemeCollectionSpec,
+    ThemeContextKeys,
+    ThemeSpec,
+} from "../types";
+import { BookingSgColourSet } from "./specs/bookingsg-color-set";
+import { CCubeColourSet } from "./specs/ccube-color-set";
+import { LifeSgColourSet } from "./specs/lifesg-color-set";
+import { MyLegacyColourSet } from "./specs/mylegacy-color-set";
+import { RBSColourSet } from "./specs/rbs-color-set";
+import { getCollection, getValue } from "../helpers";
+
+const ColourSpec: ThemeCollectionSpec<ColourCollectionsMap, ColourScheme> = {
+    collections: {
+        lifesg: LifeSgColourSet,
+        bookingsg: BookingSgColourSet,
+        rbs: RBSColourSet,
+        mylegacy: MyLegacyColourSet,
+        ccube: CCubeColourSet,
+    },
+    defaultValue: "lifesg",
+};
+
+export const getPrimitiveColour = (key: keyof ColourSet) => {
+    return (props: any): string => {
+        const theme = props.theme as ThemeSpec;
+        const colorSet: ColourSet = getCollection(
```

Comment:
can explore if `getCollection` typings can be enhanced so that we don't have to explicitly type the variables each time